### PR TITLE
Instantiate backend.Repos where required

### PR DESCRIPTION
**Pull request by eseliger**
This removes the need for a global backend.Repos that uses dbconn.Global.

TODO: We don't share the cache anymore.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/codeintel/httpapi/upload_handler.go [hunk 1]
```diff
@@ -383,11 +384,11 @@ func inferIndexer(r *http.Request) (string, error) {
 // 🚨 SECURITY: It is critical to call this function after necessary authz check
 // because this function would bypass authz to for testing if the repository and
 // commit exists in Sourcegraph.
-func ensureRepoAndCommitExist(ctx context.Context, w http.ResponseWriter, repoName, commit string) (*types.Repo, bool) {
+func ensureRepoAndCommitExist(ctx context.Context, db dbutil.DB, w http.ResponseWriter, repoName, commit string) (*types.Repo, bool) {
 	// This function won't be able to see all repositories without bypassing authz.
 	ctx = actor.WithInternalActor(ctx)
 
-	repo, err := backend.Repos.GetByName(ctx, api.RepoName(repoName))
+	repo, err := backend.NewRepos(db).GetByName(ctx, api.RepoName(repoName))
```
Comment (efritz) [inline enterprise/cmd/frontend/internal/codeintel/httpapi/upload_handler.go:391]: Could we instantiate this just once in the function?

File: enterprise/cmd/precise-code-intel-worker/internal/worker/handler.go [hunk 2]
```diff
@@ -192,12 +192,12 @@ const CloneInProgressDelay = time.Minute
 // If the repo is currently cloning, then we'll requeue the upload to be tried again later. This will not
 // increase the reset count of the record (so this doesn't count against the upload as a legitimate attempt).
 func requeueIfCloning(ctx context.Context, workerStore dbworkerstore.Store, upload store.Upload) (requeued bool, _ error) {
-	repo, err := backend.Repos.Get(ctx, api.RepoID(upload.RepositoryID))
+	repo, err := backend.NewRepos(workerStore.Handle().DB()).Get(ctx, api.RepoID(upload.RepositoryID))
```
Comment (efritz) [inline enterprise/cmd/precise-code-intel-worker/internal/worker/handler.go:195]: Could we instantiate this just once in the function?

File: enterprise/cmd/frontend/internal/codeintel/httpapi/mock_iface_test.go [hunk 3]
```diff
@@ -4,9 +4,9 @@ package httpapi
 
 import (
 	"context"
-	"sync"
-
 	dbstore "github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/stores/dbstore"
+	basestore "github.com/sourcegraph/sourcegraph/internal/database/basestore"
```
Comment (efritz) [inline enterprise/cmd/frontend/internal/codeintel/httpapi/mock_iface_test.go:8]: I'm not sure why this happens :/

File: enterprise/cmd/frontend/internal/codeintel/httpapi/upload_handler.go (hunk 1 above)
Comment (eseliger) [reply enterprise/cmd/frontend/internal/codeintel/httpapi/upload_handler.go:391]: absolutely, it was just a dumb search replace

File: cmd/frontend/backend/repos.go [hunk 4]
```diff
@@ -39,17 +40,25 @@ func (e ErrRepoSeeOther) Error() string {
 	return fmt.Sprintf("repo not found at this location, but might exist at %s", e.RedirectURL)
 }
 
-var Repos = &repos{
-	store: database.GlobalRepos,
-	cache: dbcache.NewDefaultRepoLister(database.GlobalRepos),
+// var Repos = &repos{
```
Comment (efritz) [inline cmd/frontend/backend/repos.go:43]: - [ ] Remove

File: cmd/frontend/graphqlbackend/repositories.go [hunk 5]
```diff
@@ -158,7 +158,7 @@ func (r *repositoryConnectionResolver) compute(ctx context.Context) ([]*types.Re
 			if opt2.LimitOffset != nil {
 				opt2.LimitOffset.Limit++
 			}
-			repos, err := backend.Repos.List(ctx, opt2)
+			repos, err := backend.NewRepos(r.db).List(ctx, opt2)
```
Comment (efritz) [inline cmd/frontend/graphqlbackend/repositories.go:161]: - [ ] Pass down shared cache

File: cmd/frontend/graphqlbackend/search.go [hunk 6]
```diff
@@ -411,7 +411,7 @@ func (r *searchResolver) resolveRepositories(ctx context.Context, opts resolveRe
 	repositoryResolver := &searchrepos.Resolver{
 		DB:               r.db,
 		Zoekt:            r.zoekt,
-		DefaultReposFunc: backend.Repos.ListDefault,
+		DefaultReposFunc: backend.NewRepos(r.db).ListDefault,
```
Comment (efritz) [inline cmd/frontend/graphqlbackend/search.go:414]: - [ ] Pass down shared cache

File: cmd/frontend/graphqlbackend/search_alert.go [hunk 7]
```diff
@@ -113,7 +113,7 @@ func (r *searchResolver) reposExist(ctx context.Context, options searchrepos.Opt
 	repositoryResolver := &searchrepos.Resolver{
 		DB:               r.db,
 		Zoekt:            r.zoekt,
-		DefaultReposFunc: backend.Repos.ListDefault,
+		DefaultReposFunc: backend.NewRepos(r.db).ListDefault,
```
Comment (efritz) [inline cmd/frontend/graphqlbackend/search_alert.go:116]: - [ ] Pass down shared cache

File: cmd/frontend/internal/app/badge.go [hunk 8]
```diff
@@ -19,7 +20,7 @@ import (
 
 // NOTE: Keep in sync with services/backend/httpapi/repo_shield.go
 func badgeValue(r *http.Request) (int, error) {
-	totalRefs, err := backend.CountGoImporters(r.Context(), routevar.ToRepo(mux.Vars(r)))
+	totalRefs, err := backend.CountGoImporters(r.Context(), dbconn.Global, routevar.ToRepo(mux.Vars(r)))
```
Comment (efritz) [inline cmd/frontend/internal/app/badge.go:23]: Global is no longer instantiated in _another_ branch, not this one, right?

File: enterprise/cmd/frontend/internal/codeintel/httpapi/iface.go [hunk 9]
```diff
@@ -4,11 +4,13 @@ import (
 	"context"
 
 	"github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/stores/dbstore"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
 )
 
 type DBStore interface {
 	Transact(ctx context.Context) (DBStore, error)
 	Done(err error) error
+	Handle() *basestore.TransactableHandle
```
Comment (efritz) [inline enterprise/cmd/frontend/internal/codeintel/httpapi/iface.go:13]: Could we nest the ShareableStore interface in this interface instead?

File: enterprise/cmd/frontend/internal/codeintel/httpapi/upload_handler.go (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/frontend/internal/codeintel/httpapi/upload_handler.go:391]: Of course, just looking at places sed could be improved :)

File: cmd/frontend/internal/app/badge.go (hunk 8 above)
Comment (eseliger) [reply cmd/frontend/internal/app/badge.go:23]: Yeah, I want to pass it down to here too, it was just quickest to get it done before the hour was up

File: cmd/frontend/internal/app/ui/handlers.go [hunk 10]
```diff
@@ -161,7 +162,7 @@ func newCommon(w http.ResponseWriter, r *http.Request, title string, serveError
 	if _, ok := mux.Vars(r)["Repo"]; ok {
 		// Common repo pages (blob, tree, etc).
 		var err error
-		common.Repo, common.CommitID, err = handlerutil.GetRepoAndRev(r.Context(), mux.Vars(r))
+		common.Repo, common.CommitID, err = handlerutil.GetRepoAndRev(r.Context(), dbconn.Global, mux.Vars(r))
```
Comment (eseliger) [inline cmd/frontend/internal/app/ui/handlers.go:165]: Not using dbconn.Global here requires a larger refactor which I'm going to do in a separate PR.
